### ISSUE TITLE
Association class syntax consistency

### ISSUE THREAD

**Issue by Nava2** [umple/umple#47]
Originally reported on Google Code with ID 131
Owned by `atojet`

---

The following logically should work. but does not

class Student {}

class CourseSection {}

associationClass Registration 
{
- Student -- \* CourseSection;
  String grade;
  }

The following works, but is not as clear (although it does allow for potentially future
ternary associations.

class Student {}

class CourseSection {}

associationClass Registration 
{
- Student;
- CourseSection;
  String grade;

}

---

Reported by @umple on 2011-06-24 20:51:27

**Comment by Nava2** [comment]
The key thing would be to make the \* Student -- \* CourseSection; a valid syntactic variant
to make programming easier.

---

Reported by @umple on 2012-01-13 23:04:44

**Comment by Nava2** [comment]
See also issue 113

---

Reported by @umple on 2012-06-15 15:15:23
- **Labels added**: Diffic-Easy

**Comment by Nava2** [comment]
Reported by @umple on 2012-09-27 19:47:18
- **Owner removed**

**Comment by Nava2** [comment]
Assigning to Tiago as a first simple problem to solve

---

Reported by @umple on 2013-05-01 01:54:28
- **Owner changed**: `atojet`

**Comment by Nava2** [comment]
Inside association classes, only [inlineAssociation] or [singleAssociationEnd] elements
were allowed. I changed the grammar to also allow [association] elements. For it to
be correct, these association elements needed to be treated differently than regular
associations inside "association {..}" blocks, for this reason and to prepare for Issue
113 I made changes to the following methods of the UmpleInternalParser class:
- analyzeAssociationClass: redundancy of code with method analyzeAssociation removed,
  it delegates to analyzeAllAssociations;
- analyzeAllAssociations: considers the possibility that the received Token is a association
  class and considers all valid syntaxes for associations, calling analyzeAssociation
  for each;
- analyzeAssociation: includes treatment for association classes, generalizing most
  tasks and delegating some responsibilities to an external method.

The main idea behind the changes is to treat association classes as regular classes
and its associations as regular associations. They will also help implementing association
between more than 2 classes in Issue 113.

I also added a test ("010_associationClassNewSyntaxAssociation.ump") to make sure this
syntax is accepted and the resulting parsing tree is correct.

---

Reported by `atojet` on 2013-05-17 19:07:43
- _Attachment: [patch-TiagoNascimento-131-AssociationClassSyntaxConsistency.diff](https://storage.googleapis.com/google-code-attachments/umple/issue-131/comment-5/patch-TiagoNascimento-131-AssociationClassSyntaxConsistency.diff)_
